01-30-shield-blocking/actors/player/Player.gd

```diff
@@ -0,0 +1,74 @@
+extends KinematicBody2D
+class_name Player
+
+onready var animated_sprite : AnimatedSprite = $AnimatedSprite
+onready var animation_player : AnimationPlayer = $AnimationPlayer
+onready var shield : Node2D = $Shield
+
+export var move_speed : = 300.0
+export var blocking_speed : = 150.0
+export var jump_force : = 500.0
+export var gravity : = 30.0
+
+var velocity : = Vector2()
+var blocking : = false
+var current_speed : = move_speed
+
+
+func _unhandled_input(event: InputEvent) -> void:
+	if event.is_action_pressed("defend"):
+		blocking = true
+		shield.visible = true
```

Comment:
You should use a setter to manage the shield, if the visibility is tied to the blocking boolean:
```gdscript
var blocking : = false setget set_blocking

func set_blocking(value : bool) -> void:
    blocking = value
    shield.visible = value
```